### PR TITLE
Support helm charts for network resource injector

### LINKED ISSUE #118: Add Helm Chart
to facilitate easy deployment and upgrades we should have a helm chart to deploy NRI in cases where it is used without sriov-network-operator.

in addition sriov-network-operator could then deploy network-resource-injector via helm if deemed needed.
(i.e have a dependent chart in sriov-network-operator helm chart)

npwg helm charts are currently hosted in:
https://github.com/k8snetworkplumbingwg/helm-charts

### PR DESCRIPTION
**Pull request by VivekThrivikraman-est**
Currently network resource injector does not support helm charts. This commit implements helm charts for nri.

Fixes: https://github.com/k8snetworkplumbingwg/network-resources-injector/issues/118

### CODE REVIEW

File: network-resources-injector/templates/auth.yaml [hunk 1]
```diff
@@ -0,0 +1,163 @@
+
```
Comment (e0ne) [inline network-resources-injector/templates/auth.yaml]: nit: please remove empty line here

File: network-resources-injector/templates/deployment.yaml [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+    spec:
+      serviceAccount: {{ .Values.serviceAccount.name }}
+      containers:
+      - name: webhook-server
+        image: {{ .Values.image.repository }}:{{ .Values.image.tag }}
+        imagePullPolicy: {{ .Values.image.pullPolicy }}
+        command:
+        - webhook
+        args:
+        - -bind-address=0.0.0.0
```
Comment (e0ne) [inline network-resources-injector/templates/deployment.yaml]: IMO, it's not secure to bind to all IP addresses. Can we bind to localhost be default?. We need to make it configurable

File: network-resources-injector/templates/deployment.yaml [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+          runAsUser: 10000
+          runAsGroup: 10000
+        volumeMounts:
+        - name: tls
+          mountPath: /etc/tls
+      volumes:
+      - name: tls
+        emptyDir: {}
+
+# For third-party certificate, use secret resource
```
Comment (e0ne) [inline network-resources-injector/templates/deployment.yaml]: We should make it configurable via values or add TODO note here to make it configurable in the future. There is no option to change this template if chart will be used from the repository

File: network-resources-injector/templates/service.yaml [hunk 4]
```diff
@@ -0,0 +1,26 @@
+
```
Comment (e0ne) [inline network-resources-injector/templates/service.yaml]: nit: please remove empty line here

File: network-resources-injector/templates/deployment.yaml [hunk 5]
```diff
@@ -0,0 +1,96 @@
...
+      serviceAccount: {{ .Values.serviceAccount.name }}
+      containers:
+      - name: webhook-server
+        image: {{ .Values.image.repository }}:{{ .Values.image.tag }}
+        imagePullPolicy: {{ .Values.image.pullPolicy }}
+        command:
+        - webhook
+        args:
+        - -bind-address=0.0.0.0
+        - -port=8443
```
Comment (e0ne) [inline network-resources-injector/templates/deployment.yaml]: should we use port number from the values?

File: network-resources-injector/values.yaml [hunk 6]
```diff
@@ -0,0 +1,57 @@
...
+  repository: network-resources-injector
+  tag: latest
+  pullPolicy: IfNotPresent
+
+service:
+  name: network-resources-injector-service
+  port: 443
+  targetPort: 8443
+
+webhookconf:
```
Comment (e0ne) [inline network-resources-injector/values.yaml:49]: nit: webhookconf => webhook?

File: network-resources-injector/templates/pdb.yaml [hunk 7]
```diff
@@ -0,0 +1,10 @@
+apiVersion: policy/v1
+kind: PodDisruptionBudget
+metadata:
+  name: {{ .Values.pdb.name }}
```
Comment (e0ne) [inline network-resources-injector/templates/pdb.yaml:4]: Do we really need this configurable?

**Review by e0ne — CHANGES_REQUESTED** [review]

File: network-resources-injector/values.yaml (hunk 6 above)
Comment (VivekThrivikraman-est) [reply network-resources-injector/values.yaml:49]: There is webhook as key in child hierarchy also, so this was to distinguish the webhook config(kind) to the actual webhook(one in webhooks in the webhookconfig). I can change it if you feel this is better.

File: network-resources-injector/templates/pdb.yaml (hunk 7 above)
Comment (VivekThrivikraman-est) [reply network-resources-injector/templates/pdb.yaml:4]: The other names are using templates(even in other non nri templates), should we only omit for this? I feel its ok to hardcode, but I was following the other templates already available. Or should we have it as " {{.Chart.Name}}+ "-pdb" " ?

File: network-resources-injector/templates/auth.yaml (hunk 1 above)
Comment (VivekThrivikraman-est) [reply network-resources-injector/templates/auth.yaml]: Done.

File: network-resources-injector/templates/deployment.yaml (hunk 2 above)
Comment (VivekThrivikraman-est) [reply network-resources-injector/templates/deployment.yaml]: Done.

File: network-resources-injector/templates/deployment.yaml (hunk 3 above)
Comment (VivekThrivikraman-est) [reply network-resources-injector/templates/deployment.yaml]: Used condition for taking appropriate config based on type of certificate.

File: network-resources-injector/templates/deployment.yaml (hunk 5 above)
Comment (VivekThrivikraman-est) [reply network-resources-injector/templates/deployment.yaml]: Done.

File: network-resources-injector/templates/service.yaml (hunk 4 above)
Comment (VivekThrivikraman-est) [reply network-resources-injector/templates/service.yaml]: Done.

File: network-resources-injector/templates/deployment.yaml (hunk 3 above)
Comment (VivekThrivikraman-est) [reply network-resources-injector/templates/deployment.yaml]: Used condition for taking appropriate config based on type of certificate.

**Review by e0ne — APPROVED** [review]
Thank for addressing my comments!

**Comment by rthakur-est** [comment]
@e0ne can this review be merged?

**Comment by e0ne** [comment]
@rthakur-est  merged

**Pull request merged**